lib/plug/adapters/cowboy/conn.ex

```diff
@@ -2,6 +2,8 @@ defmodule Plug.Adapters.Cowboy.Conn do
   @behaviour Plug.Conn.Adapter
   @moduledoc false
 
+  @statuses Application.get_env(:plug, :statuses, %{})
```

Comment:
Doing this here will require users to recompile plug if they change something here though, right? Are we ok with this? If we are, we should document the hell out of it because I feel it's very not user friendly (though I ofc understand the advantages here).